### PR TITLE
 fix update in feature add store page

### PR DESCRIPTION
**Pull request by daiHash**
## Description
fix update in feature add store page

## Closes issue(s)

If there any issue that is fixed with this PR, please attach it/them here...

## Screenshots
sample-buy-update

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [x] I have tested this code
- [ ] Add here the checks for this PR. If any please delete these checks

### CODE REVIEW

**Comment by daiHash** [comment]
@conalli 
Fix the update issue, so please take a look and if add this changes to your branch cherry-pick my commits to your branch

File: mockServer/data/flowers/flowersData.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    isActive: true,
+    isSecret: false,
+  },
+  {
+    _id: "7128d52b4bd76bf2e3a755a5",
+    name: "Lily",
+    imageURL:
+      "https://res.cloudinary.com/dyvrke6ml/image/upload/v1631103160/icons/lily_s6luhq.gif",
+    price: 8,
+    isActive: true,
```
Comment (daiHash) [inline mockServer/data/flowers/flowersData.ts:107]: Added mock data so you can test buying locally without accessing the database, so you have infinite tries and coins👍

File: src/pages/Florist/index.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+      backgroundColor: theme.palette.error.light,
+    },
+  })
+);
+
+export const Florist = () => {
+  const classes = useStyles();
+  const { userData, setUserData } = useUserState();
+  const [lastBought, setLastBought] = useState<string | null>(null);
+  const [flowerPrice, setFlowerPrice] = useState(0);
```
Comment (daiHash) [inline src/pages/Florist/index.tsx:65]: Added flowerPrice state to keep track of current buying flower Price to use when updating global state balance

File: src/pages/Florist/index.tsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+      setUserData((data) => {
+        const newFlowerCollection =
+          (data.isLoggedIn && data.flowerCollections) || [];
+
+        const currentBalance = (data.isLoggedIn && data.balance) || 0;
+        return {
+          ...data,
+          balance: currentBalance - flowerPrice,
+          flowerCollections: [...newFlowerCollection, ...[lastBought]],
```
Comment (daiHash) [inline src/pages/Florist/index.tsx:108]: updating balance and flowerCollection

File: src/pages/Florist/index.tsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      });
+      // MEMO: Conall not sure about this logic so if needed uncomment it else remove it
+      // setBought((prev) => {
+      //   if (!bought.includes(lastBought)) {
+      //     return [...prev, lastBought];
+      //   } else {
+      //     return prev;
+      //   }
+      // });
+    }
```
Comment (daiHash) [inline src/pages/Florist/index.tsx:119]: Not sure about this code so @conalli take a look at it🙏

File: src/pages/Florist/index.tsx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+    if (buyFlowerAPIState.status === "failed") {
+      setBuyFlowerError(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [buyFlowerAPIState.status, lastBought, flowerPrice]);
```
Comment (daiHash) [inline src/pages/Florist/index.tsx:125]: The issue was using global state and adding it to the deps, so changed to use the current state directly from setUserData

File: src/pages/Florist/index.tsx (hunk 4 above)
Comment (conalli) [reply src/pages/Florist/index.tsx:119]: Yeah this was just what I was using to track bought items before using the global state so I deleted it.

**Pull request closed without merging**